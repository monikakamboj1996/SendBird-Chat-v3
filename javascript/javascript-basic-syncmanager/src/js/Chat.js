import styles from '../scss/chat.scss';
import { createDivEl } from './utils';
import { SendBirdAction } from './SendBirdAction';
import { SendBirdChatEvent } from './SendBirdChatEvent';
import { ChatLeftMenu } from './ChatLeftMenu';
import { ChatTopMenu } from './components/ChatTopMenu';
import { ChatMain } from './components/ChatMain';

let instance = null;

class Chat {
  constructor() {
    if (instance) {
      return instance;
    }
    this.body = document.querySelector('.body-center');

    this.channel = null;
    this.element = null;
    this.top = null;
    this.emptyElement = this._createEmptyElement();
    this.render();
    instance = this;
  }

  _createEmptyElement() {
    const item = createDivEl({ className: styles['chat-empty'] });

    const content = createDivEl({ className: styles['empty-content'] });
    item.appendChild(content);

    const title = createDivEl({ className: styles['content-title'], content: 'WELCOME TO SAMPLE CHAT' });
    content.appendChild(title);
    const image = createDivEl({ className: styles['content-image'] });
    content.appendChild(image);
    const desc = createDivEl({
      className: styles['content-desc'],
      content:
        'Create or select a channel to chat in.\n' +
        "If you don't have a channel to participate,\n" +
        'go ahead and create your first channel now.'
    });
    content.appendChild(desc);
    return item;
  }

  renderEmptyElement() {
    this._removeChatElement();
    this.body.appendChild(this.emptyElement);
  }

  _removeEmptyElement() {
    if (this.body.contains(this.emptyElement)) {
      this.body.removeChild(this.emptyElement);
    }
  }

  _createChatElement(channel) {
    this.element = createDivEl({ className: styles['chat-root'] });

    this.top = new ChatTopMenu(channel);
    this.element.appendChild(this.top.element);

    /// reset manager when ChatMain is obsolete
    if (this.main && this.main.body && this.main.body.collection) {
      this.main.body.collection.remove();
    }
    this.main = new ChatMain(channel);
  }

  _addEventHandler() {
    const channelEvent = new SendBirdChatEvent();
    channelEvent.onTypingStatusUpdated = groupChannel => {
      if (this.channel.url === groupChannel.url) {
        this.main.updateTyping(groupChannel.getTypingMembers());
      }
    };
  }

  _renderChatElement(channel) {
    const sendbirdAction = SendBirdAction.getInstance();
    this._removeEmptyElement();
    this._removeChatElement();
    this.channel = channel;

    ChatLeftMenu.getInstance().activeChannelItem(channel.url);
    this._addEventHandler();

    sendbirdAction
      .getChannel(channel.url)
      .then(channel => {
        this.channel = channel;
        this._createChatElement(this.channel);
        this.body.appendChild(this.element);
        this.main.loadInitialMessages();
      })
      .catch(() => {
        this._createChatElement(this.channel);
        this.body.appendChild(this.element);
        this.main.loadInitialMessages();
      });
  }

  _removeChatElement() {
    const chatElements = this.body.getElementsByClassName(styles['chat-root']);
    Array.prototype.slice.call(chatElements).forEach(chatEl => {
      chatEl.parentNode.removeChild(chatEl);
    });
  }

  render(channel) {
    channel ? this._renderChatElement(channel) : this.renderEmptyElement();
  }

  static getInstance() {
    return new Chat();
  }
}

export { Chat };
