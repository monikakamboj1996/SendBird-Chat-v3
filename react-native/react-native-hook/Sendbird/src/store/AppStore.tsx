import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
