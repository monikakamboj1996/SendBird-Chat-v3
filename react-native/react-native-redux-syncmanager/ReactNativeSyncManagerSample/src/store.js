import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
