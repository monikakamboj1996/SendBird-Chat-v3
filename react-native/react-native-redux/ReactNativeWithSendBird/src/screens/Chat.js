import React, { Component } from 'react';
import { Platform, View, FlatList, Text, Alert, BackHandler, Clipboard, Keyboard } from 'react-native';
import { NavigationActions } from 'react-navigation';
import Permissions, { PERMISSIONS, RESULTS } from 'react-native-permissions';
import { connect } from 'react-redux';
import {
  openChannelProgress,
  groupChannelProgress,
  initChatScreen,
  getChannelTitle,
  createChatHandler,
  onSendButtonPress,
  getPrevMessageList,
  onUserBlockPress,
  onMessageDelete,
  onFileButtonPress,
  typingStart,
  onUserMessageCopy,
  onUserUpdateMessage,
  clearMessageSelection,
  typingEnd,
  channelExit
} from '../actions';
import { Button, Spinner, MessageInput, Message, AdminMessage } from '../components';
import { BarIndicator } from 'react-native-indicators';
import ImagePicker from 'react-native-image-picker';
import {
  sbGetGroupChannel,
  sbGetOpenChannel,
  sbCreatePreviousMessageListQuery,
  sbAdjustMessageList,
  sbMarkAsRead
} from '../sendbirdActions';
import appStateChangeHandler from '../appStateChangeHandler';
import SendBird from 'sendbird';

class Chat extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state;
    let newTitle = '';
    if (params.chooseTitle) {
      newTitle = params.chooseTitle;
    } else {
      newTitle = `${params.newTitle}` + (params.memberCount ? `(${params.memberCount})` : null);
    }
    let headerLeft = this._defaultHeaderLeft(params);
    if (navigation.getParam('configurableHeaderLeft')) {
      headerLeft = navigation.getParam('configurableHeaderLeft');
    }
    let headerRight = this._defaultHeaderRight(params, navigation);
    if (navigation.getParam('configurableHeaderRight')) {
      headerRight = navigation.getParam('configurableHeaderRight');
    }
    return {
      title: newTitle,
      headerLeft: headerLeft,
      headerRight: headerRight
    };
  };

  static _defaultHeaderLeft = params => {
    return (
      <Button
        containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
        buttonStyle={{ paddingLeft: 14 }}
        icon={{
          name: 'chevron-left',
          type: 'font-awesome',
          color: '#7d62d9',
          size: 18
        }}
        backgroundColor="transparent"
        onPress={() => {
          params.handleHeaderLeft();
        }}
      />
    );
  };

  static _defaultHeaderRight = (params, navigation) => {
    const _renderInviteButton = () => {
      return params.isOpenChannel ? null : (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 0, paddingRight: 0 }}
          iconRight={{
            name: 'user-plus',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => {
            navigation.navigate('GroupChannelInvite', {
              title: 'Invite',
              channelUrl: params.channelUrl
            });
          }}
        />
      );
    };

    return (
      <View style={{ flexDirection: 'row' }}>
        {_renderInviteButton()}
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 4, paddingRight: 4 }}
          iconRight={{
            name: 'users',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => {
            navigation.navigate('Member', {
              isOpenChannel: params.isOpenChannel,
              channelUrl: params.channelUrl
            });
          }}
        />
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 0, paddingRight: 14 }}
          iconRight={{
            name: 'user-times',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => {
            navigation.navigate('BlockUser');
          }}
        />
      </View>
    );
  };

  constructor(props) {
    super(props);
    this.flatList = null;
    this.state = {
      channel: null,
      isLoading: false,
      previousMessageListQuery: null,
      textMessage: ''
    };
  }

  componentDidMount() {
    this.willFocusSubsription = this.props.navigation.addListener('willFocus', () => {
      this._init();
    });
    this.appStateHandler = appStateChangeHandler.getInstance().addCallback('CHAT', () => {
      this._init();
    });
    this.props.navigation.setParams({
      handleHeaderLeft: this._onBackButtonPress
    });
    BackHandler.addEventListener('hardwareBackPress', this._onBackButtonPress);
    this._init();
  }

  componentWillUnmount() {
    this.appStateHandler();
    this.willFocusSubsription.remove();
    BackHandler.removeEventListener('hardwareBackPress', this._onBackButtonPress);
  }

  _init = () => {
    this.props.initChatScreen();
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    if (isOpenChannel) {
      sbGetOpenChannel(channelUrl).then(channel => this.setState({ channel }, () => this._componentInit()));
    } else {
      sbGetGroupChannel(channelUrl).then(channel => this.setState({ channel }, () => this._componentInit()));
    }
  };

  _componentInit = () => {
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    this.props.openChannelProgress(false);
    this.props.groupChannelProgress(false);
    this.props.getChannelTitle(channelUrl, isOpenChannel);
    this.props.createChatHandler(channelUrl, isOpenChannel);
    this._getMessageList(true);
    if (!isOpenChannel) {
      sbMarkAsRead({ channelUrl });
    }
  };

  componentDidUpdate(prevProps) {
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    const { title, memberCount, list, exit, selectedMessages } = this.props;

    if (!isOpenChannel) {
      this.state.textMessage ? this.props.typingStart(channelUrl) : this.props.typingEnd(channelUrl);
    }

    if (memberCount !== prevProps.memberCount || title !== prevProps.title) {
      const setParamsAction = NavigationActions.setParams({
        params: { memberCount, newTitle: title },
        key: this.props.navigation.state.key
      });
      this.props.navigation.dispatch(setParamsAction);
    }

    const { params } = this.props.navigation.state;
    if (
      selectedMessages &&
      selectedMessages.length !== 0 &&
      (params.chooseTitle !== 'Choose' || selectedMessages !== prevProps.selectedMessages)
    ) {
      this._updateHeadersOnMessagePress(selectedMessages);
    }
    if ((!selectedMessages || selectedMessages.length === 0) && params.chooseTitle === 'Choose') {
      this.props.navigation.setParams({
        configurableHeaderLeft: undefined,
        configurableHeaderRight: undefined,
        chooseTitle: null,
        newTitle: title
      });
    }

    if (list !== prevProps.list) {
      this.setState({ isLoading: false });
    }

    if (exit && prevProps.exit !== exit) {
      this.setState({ isLoading: false }, () => {
        this.props.navigation.goBack();
      });
    }
  }

  _onBackButtonPress = () => {
    if (this.props.selectedMessages && this.props.selectedMessages.length !== 0) {
      this._onEditingCancel();
      return this.props.clearMessageSelection();
    }
    const { channelUrl, isOpenChannel, _initListState } = this.props.navigation.state.params;
    if (_initListState) _initListState();
    this.setState({ isLoading: true }, () => {
      this.props.channelExit(channelUrl, isOpenChannel);
    });
    return true;
  };

  _onTextMessageChanged = textMessage => {
    this.setState({ textMessage });
  };

  _onUserBlockPress = userId => {
    Alert.alert('User Block', 'Are you sure want to block user?', [
      { text: 'Cancel' },
      {
        text: 'OK',
        onPress: () => this.props.onUserBlockPress(userId)
      }
    ]);
  };

  _getMessageList = init => {
    if (!this.state.previousMessageListQuery && !init) {
      return;
    }
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    this.setState({ isLoading: true }, () => {
      if (init) {
        sbCreatePreviousMessageListQuery(channelUrl, isOpenChannel)
          .then(previousMessageListQuery => {
            this.setState({ previousMessageListQuery }, () => {
              this.props.getPrevMessageList(this.state.previousMessageListQuery);
            });
          })
          .catch(error => this.props.navigation.goBack());
      } else {
        this.props.getPrevMessageList(this.state.previousMessageListQuery);
      }
    });
  };

  _onSendButtonPress = () => {
    if (this.state.textMessage) {
      const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
      const { textMessage } = this.state;
      this.setState({ textMessage: '' }, () => {
        this.props.onSendButtonPress(channelUrl, isOpenChannel, textMessage);
        if (this.props && this.props.list && this.props.list.length > 0) {
          this.flatList.scrollToIndex({
            index: 0,
            viewOffset: 0
          });
        }
      });
    }
  };

  _onPhotoAddPress = () => {
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    Permissions.check(
      Platform.OS === 'ios' ? PERMISSIONS.IOS.PHOTO_LIBRARY : PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
    ).then(response => {
      if (response === RESULTS.GRANTED) {
        ImagePicker.showImagePicker(
          {
            title: 'Select Image File To Send',
            mediaType: 'photo',
            noData: true
          },
          response => {
            if (!response.didCancel && !response.error && !response.customButton) {
              let source = { uri: response.uri };
              if (response.fileName) {
                source['name'] = response.fileName;
              } else {
                let paths = response.uri.split('/');
                source['name'] = paths[paths.length - 1];
              }
              if (response.type) {
                source['type'] = response.type;
              } else {
                /** For react-native-image-picker library doesn't return type in iOS,
                 *  it is necessary to force the type to be an image/jpeg (or whatever you're intended to be).
                 */
                if (Platform.OS === 'ios') {
                  source['type'] = 'image/jpeg';
                }
              }
              this.props.onFileButtonPress(channelUrl, isOpenChannel, source);
            }
          }
        );
      } else if (response === RESULTS.DENIED) {
        Permissions.request(
          Platform.OS === 'ios' ? PERMISSIONS.IOS.PHOTO_LIBRARY : PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
        ).then(response => {
          this._onPhotoAddPress();
        });
      } else {
        Alert.alert('Permission denied', 'You declined the permission to access to your photo.', [{ text: 'OK' }], {
          cancelable: false
        });
      }
    });
  };

  _selectMessages = messages => {
    const sb = SendBird.getInstance();
    return (
      <View style={{ flexDirection: 'row' }}>
        {sb.currentUser.userId === messages[0].sender.userId && (
          <Button
            containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
            buttonStyle={{ paddingLeft: 10, paddingRight: 10 }}
            iconRight={{
              name: 'edit',
              type: 'font-awesome',
              color: '#7d62d9',
              size: 22
            }}
            backgroundColor="transparent"
            onPress={() => {
              if (this.props.selectedMessages[0].isFileMessage() || this.props.selectedMessages[0].isAdminMessage()) {
                return;
              }
              this.setState({
                textMessage: this.props.selectedMessages[0].message,
                editing: true
              });
            }}
          />
        )}
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 10, paddingRight: 10 }}
          iconRight={{
            name: 'copy',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 22
          }}
          backgroundColor="transparent"
          onPress={() => {
            if (this.props.selectedMessages[0].isFileMessage()) {
              return;
            }
            this.props.onUserMessageCopy();
            Clipboard.setString(this.props.selectedMessages[0].message);
          }}
        />
        {sb.currentUser.userId === messages[0].sender.userId && (
          <Button
            containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
            buttonStyle={{ paddingLeft: 10, paddingRight: 14 }}
            iconRight={{
              name: 'trash',
              type: 'font-awesome',
              color: '#7d62d9',
              size: 22
            }}
            backgroundColor="transparent"
            onPress={() => {
              this._deleteSelectedMessageAlert(this.props.selectedMessages[0]);
            }}
          />
        )}
      </View>
    );
  };

  _deleteSelectedMessageAlert = message => {
    if (message.isAdminMessage()) {
      return;
    }
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    Alert.alert('Delete Message', 'Are you sure want to delete message?', [
      { text: 'Cancel' },
      {
        text: 'OK',
        onPress: () => this.props.onMessageDelete(channelUrl, isOpenChannel, message)
      }
    ]);
  };

  _updateHeadersOnMessagePress = messages => {
    const sb = SendBird.getInstance();
    const configurableHeaderRight = this._selectMessages(messages);
    this.props.navigation.setParams({
      chooseTitle: 'Choose',
      configurableHeaderLeft: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 14 }}
          icon={{
            name: 'times',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 22
          }}
          backgroundColor="transparent"
          onPress={() => {
            this.props.clearMessageSelection();
          }}
        />
      ),
      configurableHeaderRight: configurableHeaderRight
    });
  };

  _renderList = rowData => {
    const message = rowData.item;
    const { isOpenChannel } = this.props.navigation.state.params;
    const { channel } = this.state;
    if (message.isUserMessage() || message.isFileMessage()) {
      return (
        <Message
          key={message.messageId ? message.messageId : message.reqId}
          isShow={message.sender.isShow}
          isUser={message.isUser}
          profileUrl={message.sender.profileUrl.replace('http://', 'https://')}
          onAvatarPress={() => this._onUserBlockPress(message.sender.userId)}
          nickname={message.sender.nickname}
          time={message.time}
          isEdited={message.isEdited}
          readCount={isOpenChannel || !channel ? 0 : channel.getReadReceipt(message)}
          message={message}
        />
      );
    } else if (message.isAdminMessage()) {
      return <AdminMessage message={message.message} />;
    } else {
      return <View />;
    }
  };

  _renderTyping = () => {
    const { isOpenChannel } = this.props.navigation.state.params;
    return isOpenChannel ? null : (
      <View style={styles.renderTypingViewStyle}>
        <View style={{ opacity: this.props.typing ? 1 : 0, marginRight: 8 }}>
          <BarIndicator count={4} size={10} animationDuration={900} color="#cbd0da" />
        </View>
        <Text style={{ color: '#cbd0da', fontSize: 10 }}>{this.props.typing}</Text>
      </View>
    );
  };

  _onEditingCancel = () => {
    this.setState({
      textMessage: '',
      editing: false
    });
    Keyboard.dismiss();
  };

  _onEditingApprove = () => {
    const { channelUrl, isOpenChannel } = this.props.navigation.state.params;
    this.props.onUserUpdateMessage(channelUrl, isOpenChannel, this.props.selectedMessages[0], this.state.textMessage);
    this._onEditingCancel();
  };

  render() {
    return (
      <View style={styles.containerViewStyle}>
        <Spinner visible={this.state.isLoading} />
        <View style={styles.messageListViewStyle}>
          <FlatList
            ref={elem => (this.flatList = elem)}
            renderItem={this._renderList}
            data={this.props.list}
            extraData={this.state}
            keyExtractor={(item, index) => item.messageId + ''}
            onEndReached={() => this._getMessageList(false)}
            onEndReachedThreshold={0}
          />
        </View>
        <View style={styles.messageInputViewStyle}>
          {this._renderTyping()}
          <MessageInput
            onLeftPress={this.state.editing ? this._onEditingCancel : this._onPhotoAddPress}
            onRightPress={this.state.editing ? this._onEditingApprove : this._onSendButtonPress}
            editing={this.state.editing}
            textMessage={this.state.textMessage}
            onChangeText={this._onTextMessageChanged}
          />
        </View>
      </View>
    );
  }
}

function mapStateToProps({ chat }) {
  let { title, memberCount, list, exit, typing, selectedMessages } = chat;

  list = sbAdjustMessageList(list);
  return { title, memberCount, list, exit, typing, selectedMessages };
}

export default connect(
  mapStateToProps,
  {
    openChannelProgress,
    groupChannelProgress,
    initChatScreen,
    getChannelTitle,
    createChatHandler,
    onSendButtonPress,
    getPrevMessageList,
    onUserUpdateMessage,
    onUserBlockPress,
    onFileButtonPress,
    onUserMessageCopy,
    clearMessageSelection,
    typingStart,
    typingEnd,
    onMessageDelete,
    channelExit
  }
)(Chat);

const styles = {
  renderTypingViewStyle: {
    flexDirection: 'row',
    marginLeft: 14,
    marginRight: 14,
    marginTop: 4,
    marginBottom: 0,
    paddingBottom: 0,
    height: 14
  },
  containerViewStyle: {
    backgroundColor: '#f1f2f6',
    flex: 1
  },
  messageListViewStyle: {
    flex: 10,
    transform: [{ scaleY: -1 }]
  },
  messageInputViewStyle: {
    flex: 1,
    marginBottom: 0,
    flexDirection: 'column',
    justifyContent: 'center'
  }
};
