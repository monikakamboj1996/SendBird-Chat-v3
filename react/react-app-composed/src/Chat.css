.sendbird-app__wrap {
  width: 100%;
  height: 100%;
  display: flex;
}
.sendbird-app__channellist-wrap {
  max-width: 320px;
}
.sendbird-app__conversation-wrap {
  flex: 1;
  position: relative;
}
.sendbird-app__settingspanel-wrap {
  position: fixed;
  top: 0;
  right: 0;
  height: 100%;
}
