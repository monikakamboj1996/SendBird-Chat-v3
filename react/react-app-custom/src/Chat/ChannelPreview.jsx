import React from 'react';

import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';

export default function ChannelPreview({ channel, onLeaveChannel }) {
  return (
    <Paper variant="outlined" style={{ display: 'flex' }}>
      <div style={{
        display: 'flex',
        flexDirection: 'column',
      }}>
        <CardContent style={{ flex: '1 0 auto', minWidth: 180 }}>
          <Typography component="h5" variant="h5">
            {channel.name
             || channel.members.map(c => c.nickname || c.userId).join(', ').slice(0, 10).concat('...')
            }
          </Typography>
        </CardContent>
        <div className={{
          display: 'flex',
          alignItems: 'center',
        }}>
          <Button
            variant="outlined"
            color="secondary"
            onClick={() => onLeaveChannel(channel, () => {
              alert(`Left channel: ${channel.url}`)
            })}
          >
            Leave
          </Button>
        </div>
      </div>
      <CardMedia
        image={
          channel.coverUrl
          || channel.members[0].profileUrl
        }
        style={{ width: 151 }}
        title={channel.url}
      />
    </Paper>
  )
}
