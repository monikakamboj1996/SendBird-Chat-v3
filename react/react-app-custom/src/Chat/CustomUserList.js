class CustomPaginatedQuery {
  constructor() {
    // Required public property to determine if more data is available
    this.hasNext = true;
  }

  // Required public property
  next(callback) {
    // make async call and a get list of users
    const users = [
      {
        userId: 'sendbird',
        nickName: 'sendbird-display-name',
        profileUrl: 'https://static.sendbird.com/sample/profiles/profile_10_512px.png'
      },
      {
        userId: 'hoon100',
        nickName: 'hoon100-display-name',
        profileUrl: 'https://static.sendbird.com/sample/profiles/profile_11_512px.png'
      },
      {
        userId: 'hoon101',
        nickName: 'hoon101-display-name',
        profileUrl: 'https://static.sendbird.com/sample/profiles/profile_12_512px.png'
      },
      {
        userId: 'hoon102',
        nickName: 'hoon102-display-name',
        profileUrl: 'https://static.sendbird.com/sample/profiles/profile_13_512px.png'
      },
    ];
    const error = false;
    // set this.hasNext
    this.hasNext = false;
    callback(users, error);
  }
}

export default () => new CustomPaginatedQuery();
