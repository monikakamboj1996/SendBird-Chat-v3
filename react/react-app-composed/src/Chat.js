import React, { useEffect, useState } from 'react';

import './Chat.css';

import { useHistory } from "react-router-dom";
import {
  SendBirdProvider,
  ChannelList,
  Channel,
  ChannelSettings,
} from 'sendbird-uikit';
import 'sendbird-uikit/dist/index.css';

export default function Chat({ userId, nickname, theme }) {
  const history = useHistory();
  useEffect(() => {
    if (!userId || !nickname) {
      history.push('/');
    }
  }, [userId, nickname, history]);
  const [showSettings, setShowSettings] = useState(false);
  const [currentChannelUrl, setCurrentChannelUrl] = useState(null);
  return (
    <div style={{ height: '100vh' }}>
      <SendBirdProvider
        appId={process.env.APP_ID}
        theme={theme}
        userId={userId}
        nickname={nickname}
      >
        <div className="sendbird-app__wrap">
          <div className="sendbird-app__channellist-wrap">
            <ChannelList
              onChannelSelect={(channel) => {
                if (channel && channel.url) {
                  setCurrentChannelUrl(channel.url);
                }
              }}
            />
          </div>
          <div className="sendbird-app__conversation-wrap">
            <Channel
              channelUrl={currentChannelUrl}
              onChatHeaderActionClick={() => { setShowSettings(true); }}
            />
          </div>
        </div>
        {showSettings && (
          <div className="sendbird-app__settingspanel-wrap">
            <ChannelSettings
              channelUrl={currentChannelUrl}
              onCloseClick={() => { setShowSettings(false); }}
            />
          </div>
        )}
      </SendBirdProvider>
    </div>
  )
}
