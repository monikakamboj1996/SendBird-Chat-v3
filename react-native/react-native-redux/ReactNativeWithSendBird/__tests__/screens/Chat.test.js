jest.useFakeTimers();

import 'react-native';
import React from 'react';
import Chat from '../../src/screens/Chat';
import store from '../../src/store';
import renderer from 'react-test-renderer';

const navigation = {
  setParams: jest.fn(),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl',
      isOpenChannel: true
    }
  },
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('screen/Chat', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Chat store={store} navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
