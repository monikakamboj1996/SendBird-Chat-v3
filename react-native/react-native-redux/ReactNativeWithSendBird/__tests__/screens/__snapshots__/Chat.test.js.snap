// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/Chat renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f1f2f6",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 10,
        "transform": Array [
          Object {
            "scaleY": -1,
          },
        ],
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "channel": null,
          "isLoading": false,
          "previousMessageListQuery": null,
          "textMessage": "",
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 10,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#494e57",
                "fontSize": 20,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "margin": 0,
                  "padding": 0,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={[Function]}
          placeholder="Your message"
          rejectResponderTermination={true}
          selectionColor="#212529"
          style={
            Object {
              "color": "#212529",
              "minHeight": 36,
              "width": 674,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 0,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#494e57",
                "fontSize": 20,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "margin": 0,
                  "padding": 0,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
